src/net/rpc.rs

```diff
@@ -1509,6 +1510,91 @@ impl ConversationHttp {
         return response.send(http, fd).map(|_| ());
     }
 
+    /// Handle a GET health.
+    /// If the node's height is not up to the max height of all of its bootstrap nodes, we return
+    /// false for the health check.
+    /// Additionally return what percent of the max height the node height is at.
+    fn handle_get_health<W: Write>(
+        http: &mut StacksHttp,
+        fd: &mut W,
+        req: &HttpRequestType,
+        network_id: u32,
+        sortdb: &SortitionDB,
+        chainstate: &StacksChainState,
+        peerdb: &PeerDB,
+        block_stats: Option<&HashMap<NeighborKey, NeighborBlockStats>>,
+    ) -> Result<(), net_error> {
+        let response_metadata = HttpResponseMetadata::from(req);
+
+        let block_height = match chainstate.get_stacks_chain_tip(sortdb)? {
+            Some(tip) => tip.height,
+            None => 0,
+        };
+
+        let initial_neighbors =
+            PeerDB::get_valid_initial_neighbors(peerdb.conn(), network_id, block_height).unwrap();
+
+        // no bootstrap nodes found, unable to determine health.
+        if initial_neighbors.len() == 0 {
+            let response = HttpResponseType::GetHealthQueryError(
+                response_metadata.clone(),
+                "No viable bootstrap peers found, unable to determine health".to_string(),
+            );
+            return response.send(http, fd).map(|_| ());
+        }
+
+        // get max block height amongst bootstrap nodes
+        let response = if let Some(block_stats) = block_stats {
+            let mut max_height: u64 = 1;
+            let mut stats_obtained = false;
+            for neighbor in initial_neighbors.iter() {
+                let nk = &neighbor.addr;
+                match block_stats.get(nk) {
+                    Some(neighbor_block_stats) => {
+                        if neighbor_block_stats.status == NodeStatus::Online {
```

Comment:
Should we add a helper method in `NeighborBlockStats`?

Reply:
For what logic do you think we should add a helper method (it's not clear to me)?

Reply:
The "online-ness" of a node for the purposes of deciding whether or not it qualifies as a valid bootstrap peer is not represented by `neighbor_block_stats.status`.  This field represents what the local node tentatively believes about the status of the remote node.  When bootstrapping, a local node will occasionally think that its remote node has diverged from it -- i.e. it is online, but it reports a different consensus hash.  This can happen if the local node manages to start processing the sortitions in reward cycle N+1 before it has processed the PoX anchor block for reward cycle N.  Moreover, the way this situation is handled depends on whether or not the local node is doing the initial block download -- if so, then the remote node's connection is preserved, but if not, then it's severed.

What would happen from the vantage point of this health endpoint is that a node's bootstrap peers might suddenly appear to disappear, since none of them are "online" anymore.  This probably isn't what you want to report.  Instead, you might want to add a method like this:

```rust
pub fn is_peer_healthy(&self, ibd: bool) -> bool {
   // apply de Morgan's law as you see fit
   (ibd && (self.status == NodeStatus::Online ||  self.status == NodeStatus::Diverged)) || (!ibd && self.status == NodeStatus::Online)
}
```